lib/ansible/modules/cloud/docker/docker_node.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/python
+#
+# (c) 2019 Piotr Wojciechowski <piotr@it-playground.pl>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: docker_node
+short_description: Manage Docker Swarm node
+version_added: "2.8"
+description:
+    - Manages the Docker nodes via Swarm Manager
+    - Change node role
+    - Change node availability
+    - Change or remove node labels
+options:
+    state:
```

Comment:
If it only has one value (and there's no obvious extension coming up), please remove it.

Reply:
Re-adding state is trivial (by using the correct `default` value).

Reply:
> An alternative approach might be using the `state` instead of the `availability` - all module parameters would be optional and those not specified won't be changed in the upate.

I don't understand what you want to say with this. This here: "all module parameters would be optional and those not specified won't be changed in the upate" is what I would expect from docker_* modules.

Reply:
I just meant changing `availability` to `state` but that may be a little confusing because in `docker_swarm` the `state` values of `join`, `absent` and `remove` are about node membership in the cluster, and here the 'state' would mean just availability.

Reply:
Then it's probably better to keep `availability` as it is.